src/lib/storage/dictionary_segment.cpp

```diff
@@ -98,26 +98,30 @@ const T DictionarySegment<T>::value_of_value_id(const ValueID value_id) const {
 
 template <typename T>
 ValueID DictionarySegment<T>::lower_bound(const T value) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto lower_bound_iterator = std::lower_bound(_dictionary.begin(), _dictionary.end(), value);
+  if (lower_bound_iterator == _dictionary.end()) {
+    return INVALID_VALUE_ID;
+  }
+  return ValueID{(u_int32_t)std::distance(_dictionary.begin(), lower_bound_iterator)};
```

Comment:
Why do you cast to `u_int32_t`?

Reply:
Should this be changed with the `FixedWidthIntegerVector`?

Reply:
No, because this refers to the datatype used vor ValueIDs , which right now are u_ints32_t. However it might make sense to not fix this here but to just automatically cast to the int type ValueID uses, but I am not sure how to do this.